src/Glob.jl

```diff
@@ -226,20 +248,20 @@ function _match(pat::AbstractString, i0, c::Char, caseless::Bool, extended::Bool
                 return (i0, false, c=='[')
             end
         elseif extended & (mc == '\\')
-            if done(pat, i)
+            if i > ncodeunits(pat)
```

Comment:
similarly for all of these lines, use the assignment to the temporary `next` variable pattern for these that follow here